### DISCUSSION TITLE
Seperate Image for Webvault

### DISCUSSION THREAD

**Discussion by acul009** [dani-garcia/vaultwarden#3172] [Ideas]
Hi,
first of all, this is an amazing project and I really enjoy using Vaultwarden.
But I'd be interested in a simple enhancement: I would like a docker image based on caddy or nginx for just the webvault.

There may already be one, if so, please just give me a quick tip or link.

I know about the official Bitwarden vault, but I would like to use the Vaultwarden one for ensuring compatibility.

Why do I want this? I want to prevent a theoretical scenario, in which data is compromised by replacing the web vault files in a compromised container.
By making it possible to separate these two, that should no longer be possible without compromising another container.

If the idea is stupid, feel free to tell me so. I'm no security professional and sometimes unnecessarily paranoid.

**Comment by BlackDex** [comment]
I don't think we are going to provide this. For one, it adds extra a maintenance burden. Second, it makes the whole setup harder then it should be. Even Bitwarden is building a single container these days.

Also, it wouldn't help in security i think. Why would you think someone isn't able to compromise container x but could container y?
Having more containers could even increase the attack surface.

My suggest would be to use the Alpine image which has less packages installed, and those who are installed are newer versions then the debian images.

Also, you would in the end need 3 containers. One reverse proxy, Vaultwarden and web-vault.

**Reply by acul009** [reply]
Yeah, you probably have a point.
In my case I already have traefik as a reverse proxy in front and I was thinking that a potential exploit for vaultwarden would not apply to a webserver like caddy or nginx.

Though thinking about it again, a general webserver like these is probably more likely to have an active exploit than vaultwarden is.

Thanks for your feedback and suggestion. I'll just continue using vaultwarden as is.
And also thank you for your amazing project. I've been having a blast with it and quite a few of my friends are happily using my vaultwarden server too.
